### PR TITLE
First pass at limiting moon data request frequency

### PR DESCRIPTION
**Pull request by rcuhljr**
Only request moon data near expected events.

### CODE REVIEW

File: dependency.lic [hunk 1]
```diff
@@ -1022,8 +1025,18 @@ class MoonWatch
     @queue << [moon, data]
   end
 
+  def snapshot_seconds_until_load
+    return 60 unless UserVars.moons.first['timer'] && UserVars.sun['timer']
```
Comment (jandersson) [inline dependency.lic:1029]:
```
>                                                                   
--- Lich: error: no implicit conversion of String into Integer      
        dependency:1029:in `[]'                                     
        dependency:1029:in `snapshot_seconds_until_load'            
--- Lich: dependency has exited.                                    
```

Comment (jandersson) [reply dependency.lic:1029]:
```
>;e echo UserVars.moons.first['timer']                         
--- Lich: exec1 active.                                        
--- Exception: no implicit conversion of String into Integer   
exec1:1:in `[]'                                                
--- Lich: exec1 has exited.                                    
>;e echo UserVars.moons.first                                  
--- Lich: exec1 active.                                        
[exec1: ["katamba", {}]]                                       
--- Lich: exec1 has exited.                                    
>;e echo UserVars.moons.first[1]['timer']                      
--- Lich: exec1 active.                                        
[exec1: ]                                                      
--- Lich: exec1 has exited.                                    
           
```
Is this what we want?

Comment (jandersson) [reply dependency.lic]: Testing this
```
return 60 unless UserVars.moons.reject { |m| m == "visible" }.any? { |_,v| v.key? 'visible' } && UserVars.sun['timer']
```

Comment (rcuhljr) [reply dependency.lic]: I pushed an update, but we can reject visible, I forgot about that guy.

**Comment by jandersson** [comment]
Seems legit so far

```
>                                                                                                                                                                                
[dependency: retrieving moon data]                                                                                                                                               
[dependency: calculting next moon data request]                                                                                                                                  
[dependency: Next request in 480.0 seconds]                                                                                                                                      
>                                                                                                                                                                                
With a slight tremor in his voice, Dane reprises the last few words of the chorus of his infectious tune, drawing out the final note of each phrase and letting it gradually fade
>moon                                                                                                                                                                            
katamba will rise in 10 minutes : yavash will rise in 96 minutes : xibar is up for 93 minutes                                                                                    
[play]>play concerto masterful on zills                                                                                                                                          
Your zills's dirtiness may affect your performance.                                                                                                                              
You begin a masterful concerto on your thin-edged zills.                                                                                                                         
>                                                                                                                                                                                
Peacemaker Sexton just arrived.                                                                                                                                                  
>                                                                                                                                                                                
Peacemaker Sexton scampers west.                                                                                                                                                 
>;e echo 480/60                                                                                                                                                                  
>                                                                                                                                                                                
--- Lich: exec1 active.                                                                                                                                                          
[exec1: 8]                                                                                                                                                                       
```

**Pull request merged**